lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -64,7 +62,7 @@ def matches?(given_proc, negative_expectation=false, &block)
 
         # @private
         def does_not_match?(given_proc)
-          prevent_invalid_expectations
+          warn_about_invalid_expectations
```

Comment:
I'm thinking "invalid" is the right term to use here anymore since we allow it and it has to do with false positives...